cmake/Dependencies.cmake

```diff
@@ -1018,9 +1018,30 @@ if(USE_MPI)
       if(_output MATCHES "smcuda")
         message(STATUS "Found OpenMPI with CUDA support built.")
       else()
-        message(WARNING "OpenMPI found, but it is not built with CUDA support.")
-        set(CAFFE2_FORCE_FALLBACK_CUDA_MPI 1)
+        if(USE_CUDA_MPI)
+            message(WARNING "OpenMPI with CUDA support not found, but forcing anyway.")
+            set(CAFFE2_USE_CUDA_MPI 1)
+        else()
+            message(WARNING "OpenMPI found, but it is not built with CUDA support.")
+            set(CAFFE2_FORCE_FALLBACK_CUDA_MPI 1)
+        endif()
       endif()
+    else()
+        find_program(MV2_INFO NAMES mpiname HINTS ${MPI_CXX_LIBRARIES}/../bin)
+        if(MV2_INFO)
+            execute_process(COMMAND ${MV2_INFO} "-a" OUTPUT_VARIABLE _output)
+            if(_output MATCHES "enable-cuda")
+                message(STATUS "Found MVAPICH2 with CUDA support built.")
+            else()
+                if(USE_CUDA_MPI)
+                    message(WARNING "MVAPICH2 with CUDA support not found, but forcing anyway.")
+                    set(CAFFE2_USE_CUDA_MPI 1)
+                else()
+                    message(WARNING "MVAPICH2 found, but it is not built with CUDA SUPPORT.")
+                    set(CAFFE2_FORCE_FALLBACK_CUDA_MPI 1)
+                endif()
```

Comment:
This looks like a duplicated code, could it perhaps be refactored to a function/macro?